src/include/update/rootpart.rb

```diff
@@ -459,6 +459,8 @@ def RootPartitionDialog(flavor)
         # the target distribution from the base product to make the new service
         # repositories compatible with the base product at upgrade (bnc#881320)
         if Pkg.TargetInitializeOptions(Installation.destdir,
+          # enable DB rebuild at upgrade (bsc#1209565)
+          "rebuild_db"    => true,
           "target_distro" => target_distribution) != true
```

Comment:
if you touch it, maybe it should change that `!= true` to simple `!Pkg.TargetInitializeOptions` so it will be easier to read code.